device/manager.go

```diff
@@ -86,9 +104,12 @@ type Manager interface {
 	Registry
 }
 
+// ManagerOption is a configuration option for a manager
+type ManagerOption func(*manager)
+
 // NewManager constructs a Manager from a set of options.  A ConnectionFactory will be
 // created from the options if one is not supplied.
-func NewManager(o *Options) Manager {
+func NewManager(o *Options, options ...ManagerOption) Manager {
```

Comment:
A simpler approach is to make `Filter` a field of the `Options` struct.  Then, move the creation semantics for the `Filter` somewhere else.  That keeps the manager code simple and easier to maintain.